src/rules/helpers.ts

```diff
@@ -53,3 +54,165 @@ export function buildRule<Name extends RuleName>({
     execute,
   };
 }
+
+/**
+ * A rule utility which determines whether a file exists in the project.
+ *
+ * @param filePath - The path to test.
+ * @param ruleExecutionArguments - Rule execution arguments.
+ * @returns Either a successful or failed result.
+ */
+export async function fileExists(
+  filePath: string,
+  ruleExecutionArguments: RuleExecutionArguments,
+): Promise<PartialRuleExecutionResult> {
+  const { project, pass, fail } = ruleExecutionArguments;
+  const stats = await project.fs.getEntryStats(filePath);
+
+  if (!stats) {
+    return fail([
+      {
+        message: `\`${filePath}\` does not exist in this project.`,
+      },
+    ]);
+  }
+
+  if (!stats.isFile()) {
+    return fail([
+      {
+        message: `\`${filePath}\` is not a file when it should be.`,
+      },
+    ]);
+  }
+
+  return pass();
+}
+
+/**
+ * A rule utility which determines whether a directory exists in the project.
+ *
+ * @param directoryPath - The path to test.
+ * @param ruleExecutionArguments - Rule execution arguments.
+ * @returns Either a successful or failed result.
+ */
+export async function directoryExists(
+  directoryPath: string,
+  ruleExecutionArguments: RuleExecutionArguments,
+): Promise<PartialRuleExecutionResult> {
+  const { project, pass, fail } = ruleExecutionArguments;
+  const stats = await project.fs.getEntryStats(directoryPath);
+
+  if (!stats) {
+    return fail([
+      {
+        message: `\`${directoryPath}/\` does not exist in this project.`,
+      },
+    ]);
+  }
+
+  if (!stats.isDirectory()) {
+    return fail([
+      {
+        message: `\`${directoryPath}/\` is not a directory when it should be.`,
+      },
+    ]);
+  }
+
+  return pass();
+}
+
+/**
+ * A rule utility which determines not only whether a file that's assumed to
+ * exist in the template exists in the project as well, but also whether it
+ * matches the same file in the template content-wise.
+ *
+ * @param filePath - The path to a file in both the template and project.
+ * @param ruleExecutionArguments - Rule execution arguments.
+ * @returns Either a successful or failed result.
+ */
+export async function fileConforms(
```

Comment:
These last three functions appear to be missing tests

Reply:
Tests added here: [b683cd0](https://github.com/MetaMask/module-lint/pull/39/commits/b683cd0656ce5966a3e022ecfd9b282ae28cda58)